File: leetcode/arrays/two_sum.js



function two_sum(nums, target) {
  const map = {}
  
  for (let i = 0; i < nums.length; i +=1) {
      const num = nums[i];
      const diff = target - num;
      console.log(diff, num)
      console.log(map)
      if (map[diff] !== undefined) return [map[diff], i];
      else map[num] = i;
  }
  console.log(map)
  return [-1, -1] 
}

console.log(two_sum([2,7,11,15], 9))

File: leetcode/arrays/two_sum.rs
use std::collections::HashMap;

fn main() {
    println!("{:?}", two_sum(vec![3, 2, 1], 4));
}
fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    let mut map: HashMap<i32, i32> = HashMap::new();
    for (i, num) in nums.iter().enumerate() {
        let diff = target - num;
        if map.contains_key(&diff) {
            return vec![map.remove(&diff).unwrap(), i as i32];
        } else {
            map.insert(*num as i32, i as i32);
        }
    }
    return vec![-1, -1];
}
